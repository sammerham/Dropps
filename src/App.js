
import './App.css';
import { useState } from 'react';
import ButtonContainer from './components/ButtonContainer';


/** App
 * 
 * Props:
 *  - none
 * 
 * State:
 *  - checked (boolean)
 * App -> { ButtonContainer }
 */

const App = () => {
  const [checked, setChecked] = useState(true);

  // fn to handle render button check.
  const handleCheck = () => setChecked(!checked);

  return (
    <div className='main-App'>
      <label htmlFor="checkbox" className='checkboxLabel'>Render button</label>
      <input
        type="checkbox"
        name="checkbox"
        id="checkbox"
        onClick={handleCheck}
        defaultChecked={true}
        className= {checked ? 'checked' : 'unchecked'}
      />
      {checked && <ButtonContainer />}
    </div>
  );
}

export default App;
