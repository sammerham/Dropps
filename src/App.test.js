import { render, screen, fireEvent} from '@testing-library/react';
import App from './App';


test('It renders without crashing', () => {
  render(<App />)
});

test('It matches snapshot', () => {
  const { container } = render(<App />);
  expect(container).toMatchSnapshot();
});

// {exact:false}

test('It should show label', () => {
  render(<App />);
  let label = screen.getByText('Render button');
  expect(label).toBeInTheDocument();
});

test('It should show checked checkBox and button container', () => {
  render(<App />);
  let checkBox = screen.getByLabelText('Render button');
  expect(checkBox).toHaveClass('checked');
  let clickMeButton = screen.getByText('Click Me');
  expect(clickMeButton).toBeInTheDocument()
  let cancelButton = screen.getByText('Cancel');
  expect(cancelButton).toBeInTheDocument();
});



test('It should not show button container when checkbox is unchecked', () => {
  render(<App />);
  let checkBox = screen.getByLabelText('Render button');
  expect(checkBox).toHaveClass('checked');
  let clickMeButton = screen.getByText('Click Me');
  expect(clickMeButton).toBeInTheDocument()
  let cancelButton = screen.getByText('Cancel');
  expect(cancelButton).toBeInTheDocument();
  fireEvent.click(checkBox);
  expect(clickMeButton).not.toBeInTheDocument()
  expect(cancelButton).not.toBeInTheDocument();
});

