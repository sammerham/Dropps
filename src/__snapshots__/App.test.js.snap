// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It matches snapshot 1`] = `
<div>
  <div
    class="main-App"
  >
    <label
      class="checkboxLabel"
      for="checkbox"
    >
      Render button
    </label>
    <input
      checked=""
      class="checked"
      id="checkbox"
      name="checkbox"
      type="checkbox"
    />
    <div>
      <button
        class="buttonContainer-btn"
      >
        Click Me
      </button>
      <button
        class="buttonContainer-btn"
        disabled=""
      >
        Cancel
      </button>
      <br />
      <div
        class="buttonContainer-div"
      >
        Renders: 
        <span
          class="buttonContainer-span"
        >
           
          1
        </span>
        <br />
        <br />
        Origin:
        <span
          class="buttonContainer-span"
        >
           
        </span>
      </div>
    </div>
  </div>
</div>
`;
